### PR TITLE
Toolchain file update

### LINKED ISSUE #216: Seach for optional modules in environment as well
There are also some calls to find utilities in the `common.cmake` file to find utilities like `arm-none-eabi-objcopy` or `arm-none-eabi-size`:

```cmake
find_program(CMAKE_OBJCOPY NAMES ${STM32_TARGET_TRIPLET}-objcopy PATHS ${TOOLCHAIN_BIN_PATH} NO_DEFAULT_PATH)
find_program(CMAKE_OBJDUMP NAMES ${STM32_TARGET_TRIPLET}-objdump PATHS ${TOOLCHAIN_BIN_PATH} NO_DEFAULT_PATH)
find_program(CMAKE_SIZE NAMES ${STM32_TARGET_TRIPLET}-size PATHS ${TOOLCHAIN_BIN_PATH} NO_DEFAULT_PATH)
find_program(CMAKE_DEBUGGER NAMES ${STM32_TARGET_TRIPLET}-gdb PATHS ${TOOLCHAIN_BIN_PATH} NO_DEFAULT_PATH)
find_program(CMAKE_CPPFILT NAMES ${STM32_TARGET_TRIPLET}-c++filt PATHS ${TOOLCHAIN_BIN_PATH} NO_DEFAULT_PATH)
```

For the necessary toolchain components, adding the binary folder to the environment is already sufficient.
I think this would be good here as well so it behaves consistently. I think removing `NO_DEFAULT_PATH` is sufficient to get this
behaviour.

### PR DESCRIPTION
**Pull request by rmspacefish**
1. Use `STM32_TOOLCHAIN_PATH` from environment if specified as environmental variable
2. Find tools like objcopy and size in default paths as well, which allows
   simply adding the binary folder of the toolchain to the path to find them.
   This also allows more consistent behaviour, because this already works
   for <cross-compile>-gcc/g++

closes #216

### CODE REVIEW

**Comment by atsju** [comment]
For information you might have a look at #218 + #211

**Comment by rmspacefish** [comment]
Ok. Maybe we can merge it first? It looks finished, but I have not read everything yet

**Comment by atsju** [comment]
I'm at work now. But looks fine, I will test ASAP and it can be merged regardless of other PR. As I said it was only for information.

**Review by Hish15 — CHANGES_REQUESTED** [review]
Here setting `STM32_TOOLCHAIN_PATH` in a env var, will make setting the value directly impossible. I think we want to check if the var is not set (using CLI or in the cmakelist) before using ENV var.

File: cmake/stm32/common.cmake [hunk 1]
```diff
@@ -10,6 +10,12 @@ foreach(FAMILY ${STM32_SUPPORTED_FAMILIES_LONG_NAME})
 endforeach()
 list(REMOVE_DUPLICATES STM32_SUPPORTED_FAMILIES_SHORT_NAME)
 
+if(DEFINED ENV{STM32_TOOLCHAIN_PATH})
+    message(STATUS "Detected toolchain path STM32_TOOLCHAIN_PATH in environmental variables: ")
+    message(STATUS "$ENV{STM32_TOOLCHAIN_PATH}")
+    set(STM32_TOOLCHAIN_PATH $ENV{STM32_TOOLCHAIN_PATH})
+endif()
+
 if(NOT STM32_TOOLCHAIN_PATH)
```
Comment (atsju) [inline cmake/stm32/common.cmake:19]:
```suggestion
if(NOT STM32_TOOLCHAIN_PATH)
    if(DEFINED ENV{STM32_TOOLCHAIN_PATH})
        message(STATUS "Detected toolchain path STM32_TOOLCHAIN_PATH in environmental variables: ")
        message(STATUS "$ENV{STM32_TOOLCHAIN_PATH}")
        set(STM32_TOOLCHAIN_PATH $ENV{STM32_TOOLCHAIN_PATH})
    endif()
```
@Hish15  is this what you recommend ? Would make sense.

Comment (Hish15) [reply cmake/stm32/common.cmake:19]: looks like it yeah 😄

**Review by atsju — APPROVED** [review]

**Review by Hish15 — APPROVED** [review]

**Pull request merged**